Assets/MRTK/Tests/PlayModeTests/ScrollViewTests.cs

```diff
@@ -864,6 +864,104 @@ public IEnumerator CanBeScrolledByTierOrIndexOrPage()
             Assert.AreEqual(4, scrollView.FirstHiddenItemIndex, "First hidden item is different from the expected");
         }
 
+        /// <summary>
+        /// Tests if far interaction with GGV pointer can engage the scroll drag.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator GGVScroll()
+        {
+            PlayModeTestUtilities.SetHandSimulationMode(HandSimulationMode.Gestures);
+
+            // Setting up a horizontal 1x1 scroll view with two pressable buttons items
+            ScrollingObjectCollection scrollView = InstantiateScrollAndChildren(out GameObject[] scrollItems,
+                                                                                2,
+                                                                                ScrollingObjectCollection.ScrollDirectionType.UpAndDown,
+                                                                                1,
+                                                                                1,
+                                                                                ScrollingObjectCollection.VelocityType.NoVelocitySnapToItem,
+                                                                                Vector3.forward,
+                                                                                Quaternion.identity);
+            float scale = 10f;
+            scrollView.transform.localScale *= scale;
+
+            TestUtilities.PlayspaceToOriginLookingForward();
+
+            // Hand positions
+            float offset = 0.001f;
+            Vector3 initialPos = new Vector3(0.13f, -0.17f, 0.5f); // Far pointer focus is on button       
+            Vector3 scrollEngagedPos = initialPos + Vector3.up * (scrollView.HandDeltaMagThreshold + scrollView.CellHeight * scale + offset);
+
+            // Interaction with child button should behave normally if scroll drag not yet engaged
+            TestHand hand = new TestHand(Handedness.Right);
+            yield return hand.Show(initialPos);
+            yield return hand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return hand.MoveTo(scrollEngagedPos);
+
+            Assert.IsTrue(scrollView.IsDragging, "Scroll drag was not triggered.");
+
+            yield return hand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            yield return hand.Hide();
+
+            PlayModeTestUtilities.SetHandSimulationMode(HandSimulationMode.Articulated);
```

Comment:
I think that this hide/open/articulated thing is to put things back into a state that other tests are expecting.

If this is the case, you might want to move stuff like this to the TearDown() function (it doesn't exist here but it could be added).

Otherwise, what could happen is if this test ever fails, it wont' run the teardown stuff down below - so other tests that run after it will likely spuriously fail for non obvious reasons.

What's the difference between seeing 1 failure vs 3-4 failures? It depends on which you happen to choose to look at first. Let's say that the issue is actually because this test failed (and then didn't clean up) - if this causes ContentClickHappensOnTouchUp to fail, it may do so in a non-obvious way, and if you happen to choose to look at ContentClickHappensOnTouchUp first, you will likely end up wasting time digging into what went wrong there (nothing went wrong!) - the error was in a different test.

Reply:
Thats right, they are also not needed since this class derive from BasePlayModeTests, all MR services are shutdown on teardown and initialize on setup.